server/chat-plugins/hosts.ts

```diff
@@ -0,0 +1,352 @@
+/**
+ * Chat plugin to help manage hosts, proxies, and datacenters
+ * Written by Annika
+ * Original /adddatacenters command written by Zarel
+ */
+
+import {Utils} from "../../lib/utils";
+import {Datacenter} from "../ip-tools";
+
+const IP_REGEX = /^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/;
+const HOST_REGEX = /^.+\..{2,}$/;
+
+export const pages: PageTable = {
+	datacenters() {
+		this.title = "Datacenters";
+		if (!this.can('globalban')) return 'Permission denied.';
+		let html = `<div class="ladder pad"><h2>Datacenters:</h2><table>`;
+		html += `<tr><th>Lowest IP address</th><th>Highest IP address</th><th>Name</th><th>Host</th></tr>`;
+		const sortedDatacenters = IPTools.datacenters;
+		sortedDatacenters.sort((a, b) => a.minIP - a.minIP);
+		for (const datacenter of sortedDatacenters) {
+			html += `<tr>`;
+			html += `<td>${IPTools.numberToIP(datacenter.minIP)}</td>`;
+			html += `<td>${IPTools.numberToIP(datacenter.maxIP)}</td>`;
+			html += Utils.html`<td>${datacenter.name}</td>`;
+			html += Utils.html`<td>${datacenter.host}</td>`;
+			html += `</tr>`;
+		}
+		html += `</table></div>`;
+		return html;
+	},
+
+	hosts(query) {
+		function ipSort(a: string, b: string) {
```

Comment:
declaring a `function` here is slightly awkward - would it make sense to move this to the top level?